packages/contracts/contracts/L1/ValidatorPool.sol

```diff
@@ -105,72 +93,196 @@ contract ValidatorPool is ReentrancyGuardUpgradeable, Semver {
     function initialize() public initializer {}
 
     function deposit() external payable nonReentrant {
-        uint256 newDeposits = depositsOf[msg.sender] + msg.value;
-        require(msg.sender != TRUSTED_VALIDATOR, "ValidatorPool: trusted validator cannot deposit");
-        require(
-            newDeposits >= MIN_DEPOSIT_AMOUNT,
-            "ValidatorPool: the total deposit is less than minimum amount"
-        );
+        _increaseBalance(msg.sender, msg.value);
+    }
+
+    function withdraw(uint256 _amount) external payable nonReentrant {
+        _decreaseBalance(msg.sender, _amount);
+        (bool success, ) = payable(msg.sender).call{ value: _amount }("");
+        require(success);
+    }
 
-        depositsOf[msg.sender] = newDeposits;
+    /**
+     * @notice Bond asset corresponding to the given output index. This function is called when submitting output.
+     *         Note that the amount to bond should be at least the minimum bonding amount,
+     *         and the output root must also be given for later output root comparison.
+     *
+     * @param _outputIndex Index of L2 checkpoint output.
+     * @param _amount      Amount of bonding.
+     * @param _expiresAt   Expiration time (in seconds) for bonding.
+     */
+    function bond(
+        uint256 _outputIndex,
+        uint128 _amount,
+        uint128 _expiresAt
+    ) external payable {
```

Comment:
Most of the payable keywords used in this project are intended to save fee by avoiding checking `msg.value == 0`. However, in a recent discussion, it was suggested that we remove the payable keyword because we don't see much benefit in using it.